src/planning/src/planning/planning.cpp

```diff
@@ -100,17 +119,25 @@ void Planning::track_map_callback(const custom_interfaces::msg::ConeArray &msg)
   auto path = triangulations_path;
   path_smoother->defaultSmoother(path);
 
-  RCLCPP_DEBUG(this->get_logger(), "Planning published %d path points", (int)path.size());
+  RCLCPP_DEBUG(rclcpp::get_logger("rclcpp"), "Planning published %i path points\n",
+               (int)path.size());
   publish_track_points(path);
 
   if (this->publishing_visualization_msgs_) {
     publish_visualization_msgs(cone_coloring->current_left_cones,
-                               cone_coloring->current_right_cones, triangulations_path, path);
+                               cone_coloring->current_right_cones, triangulations_path, path,
+                               track->get_left_cones(), track->get_right_cones());
   }
 }
 
 void Planning::vehicle_localization_callback(const custom_interfaces::msg::VehicleState &msg) {
   this->pose = Pose(msg.position.x, msg.position.y, msg.theta);
+  if (this->recieved_first_track && !this->recieved_first_pose) {
+    this->recieved_first_pose = true;
+    run_planning_algorithms();
```

Comment:
Probably no need or gain in doing this on both callbacks, as these two are almost always published in pairs. Maybe using a message filter would be good in this case